plugins/modules/keycloak_client.py

```diff
@@ -249,6 +249,7 @@
             - Type of client.
         type: str
         choices: ['openid-connect', 'saml']
+        default: openid-connect
```

Comment:
This is potentially a breaking change for idempotency. I'm not 100% sure how the module works right now (I don't use it), but I would expect it to only change the fields that the user explicitly specified.

So in case if the client had set this to `saml`, and the module is called without this option specified, the module would - with this change - modify the value to `openid-connect`. Before this change, it wouldn't do that.

Reply:
Hello, the module sends a POST to Keycloak without giving any protocol information. The client is created by Keycloak using the openid-connect protocol.
When you run the playbook a second time, the module requests the client from Keycloak and the desired client is updated with the missing attributes. Now the protocol contains openid-connect. The module PUT the resulting json and this time the secret is generated.
I guest it may break compatibility if someone created a client and then changed the protocol directly in the Keycloak UI.
Should the attribute be mandatory if state=='present' instead?

Reply:
There are other ways to solve this than breaking existing workflows by adding a default, or by making a value required which wasn't required before. You could for example adjust the code which is called on creation to pass this value if is not provided by the user.

Reply:
Ok, should we add a warning for the implicit assignation?

Reply:
I would mention it in the documentation for the option, that on creation an implicit default is used.

Reply:
It's done, I did not use the "default: openid-connect" syntax since it's not the reel "default" behavior